feature_engine/transformation/date_transformer.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class date_feature_extractor(BaseNumericalTransformer):
+    def __init__(
+        self,
+        features: Union[None, int, str, List[Union[str, int]]] = None,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
+
+        if not fearures:
+            raise ValueError(
+                "Feature list cannot be empty.Enter features to extract from dates"
+            )
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.fearures = set(fearures)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+
+        X = super().fit(X)
+
+        # check if all the variables are of type datetime.
+        if not all([pd.is_datetime64_any_dtype(X[col]) for col in X.columns]):
+            raise ValueError("Some variables are not datetime cannot transform data")
+
+        self.input_shape_ = X.shape
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
```

Comment:
look at the checks in this class for inspiration:
https://github.com/solegalli/feature_engine/blob/master/feature_engine/creation/combine_with_reference_feature.py#L257